src/Device/Driver/LX/NanoLogger.cpp

```diff
@@ -205,7 +205,7 @@ Nano::ReadFlightList(Port &port, RecordedFlightList &flight_list,
 
   env.SetProgressRange(nflights);
 
-  unsigned requested_tail = 1;
+  unsigned requested_tail = (unsigned) std::max(1, (signed) nflights - (signed) flight_list.max_size() + 1);
```

Comment:
This looks awfully complicated and obscure. Can you rephrase this in a way that's easier to understand, or at least add an explanatory code comment?

Reply:
Well, this is what happens with different backgrounds. For me this code is very easy to read, but the serious C++ things very hard :-).
But I will happily amend - would you like me to write a longer code that is easier to read instead, or just to add explanatory comments? I will add more documentation in both cases though.

Reply:
If you think the code is good, add a comment explaining what this formula does (and why it's there).